RechargeGauntlet/LoginForm.cs

```diff
@@ -30,27 +31,66 @@ private void LoginForm_Load(object sender, EventArgs e)
 
         private void BtnLogin_Click(object sender, EventArgs e)
         {
+            //Regex to check Username
             var userNameRegex = new Regex("^[a-zA-Z][a-zA-Z0-5]{5,}$");
+            //Regex to check Password
             var passwordRegex = new Regex("^(?!^[0-9]*$)(?!^[a-zA-Z]*$)^([a-zA-Z0-9]{8,20})$");
+
+            //Checking first if the username and password is valid
             if (!userNameRegex.IsMatch(TextboxUserName.Text) ||
                 !passwordRegex.IsMatch(TextboxPassword.Text))
             {
+                //Username is found not valid
                 if (!userNameRegex.IsMatch(TextboxUserName.Text))
                 {
                     MessageBox.Show("Please enter a valid User Name! **At Least 6 characters(alphabets and digits only) are required and must present alphabet and digits**");
                 }
+                //Password is found not valid
                 else if (!passwordRegex.IsMatch(TextboxPassword.Text))
                 {
                     MessageBox.Show("Please enter a valid Password!**At Least 8 and At most 20 characters(alphabets and digits only) are required and must have at least an alphabet and a digit.**");
                 }
+
                 else
                 {
-
+                    //For future validations
                 }
             }
+            //Username and password are found valid, now checking if they are correct by matching them with database existing pairs.
             else
             {
-                MessageBox.Show("Logging In!");
+                //Creating a local instance of Database context class
+                using (RechargeGauntletDataContext _context = new RechargeGauntletDataContext())
+                {                    
+                    //Checking if the given username exists in the database
+                    if(_context.Users.Count(u => u.Username == TextboxUserName.Text)!=1)
```

Comment:
Why query 2 times in the database. When username is matched fetched all information with password also the use this password to verify.